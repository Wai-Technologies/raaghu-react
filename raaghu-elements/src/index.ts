export { default as RdsButton } from "./rds-button";
export { default as RdsButtonGroup } from "./rds-button-group";
export { default as RdsInput } from "./rds-input";
export { default as RdsCheckbox } from "./rds-checkbox";
export { default as RdsCheckboxGroup } from "./rds-checkbox-group";
export { default as RdsCheckboxParentChild } from "./rds-checkbox-parent-child";
export { default as RdsAvatar } from "./rds-avatar";
export { default as RdsLabel } from "./rds-label";
export { default as RdsColorPicker } from "./rds-color-picker";
export { default as RdsAddressDetail } from "./rds-address-detail";
export { default as RdsBreadcrumb } from "./rds-breadcrumb";
export { default as RdsSelectList } from "./rds-select-list";
export { default as RdsDropdown } from "./rds-dropdown";
export { default as RdsDatePicker } from "./rds-datepicker";
export { default as RdsCounter } from "./rds-counter";
export { default as RdsCard } from "./rds-card";
export { default as RdsRange } from "./rds-range";
export { default as RdsDoubleRange } from "./rds-double-range";
export { default as RdsInputGroup } from "./rds-input-group";
export { default as RdsCarousel } from "./rds-carousel";
export { default as RdsSideNav } from "./rds-side-nav";
export { default as RdsAccordion ,RdsAccordionItem  } from "./rds-accordion";
export { default as RdsAlert } from "./rds-alert";
export { default as RdsBadge } from "./rds-badge";
export { default as RdsBanner } from "./rds-banner";
export { default as RdsIcon } from "./rds-icon";
export { Icons } from "./rds-icon/Icons";
export { default as RdsIconLabel } from "./rds-icon-label";
export { default as RdsSearch } from "./rds-search";
export { default as RdsNavbar } from "./rds-navbar";
export { default as RdsDropdownList } from "./rds-dropdown-list";
export { default as RdsFabMenu } from "./rds-fab-menu";
export { default as RdsWidget } from "./rds-widget";
export { default as RdsIllustration } from "./rds-illustration";
export { default as RdsBigNumber } from "./rds-big-number";
export { default as RdsSpinner } from "./rds-spinner";
export { default as RdsLikeDislike } from "./rds-like-dislike";
export { default as RdsTextArea } from "./rds-text-area";
export { default as RdsBenefit } from "./rds-benefit";
export { default as RdsBankCardDetail } from "./rds-bank-card-detail";
export { default as RdsStat } from "./rds-stat";
export { default as RdsNotification } from "./rds-notification";
export { default as RdsToast } from "./rds-toast";
export { default as RdsSize } from "./rds-size";
export { default as RdsRadioButton } from "./rds-radio-button";
export { default as RdsAppDetail } from "./rds-app-detail";
export { default as RdsOffcanvas } from "./rds-offcanvas";
export { default as RdsTable } from "./rds-table";
export { default as RdsModal } from "./rds-modal";
export { default as RdsNavtabs } from "./rds-navtabs";
export { default as RdsRating } from "./rds-rating";
export { default as RdsPopularPage } from "./rds-popular-page";
export { default as RdsColorSwitcher } from "./rds-color-switcher";
export { default as RdsListGroup } from "./rds-list-group";
export { default as RdsProductImage } from "./rds-product-image";
export { default as RdsPagination } from "./rds-pagination";
export { default as RdsCollapse } from "./rds-collapse";
export { default as RdsFeatureList } from "./rds-feature-list";
export { default as RdsPopover } from "./rds-popover";
export { default as RdsPrice } from "./rds-price";
export { default as RdsProgressBar } from "./rds-progress-bar";
export { default as RdsReviewCategory } from "./rds-review-category";
export { default as RdsScrollspy } from "./rds-scrollspy";
export {default as RdsStepper} from "./rds-stepper";
export {default as RdsTag } from "./rds-tag";
export { default as RdsTeamMember } from "./rds-team-member";
export { default as RdsTestimonial } from "./rds-testimonial";
export { default as RdsTooltip } from "./rds-tooltip";
export { default as RdsWebsiteMatrix } from "./rds-website-matrix";
export { default as RdsTextEditor } from "./rds-text-editor";
export { default as RdsFileUploader } from "./rds-file-uploader";
export { default as RdsToggle } from "./rds-toggle";
export { default as RdsVideoPlayer } from "./rds-video-player";
export { default as RdsPaymentSummary } from "./rds-payment-summary";
export { default as RdsPlandiscount } from "./rds-plan-discount";
export { default as RdsWidgetPortal } from "./rds-widget-portal";
export { default as RdsTabGroup } from "./rds-tab-group";
// export { default as RdsAnimation } from "./rds-animation";
// //Charts
export { default as RdsAreaChart } from "./rds-chart-area";
export { default as RdsLineChart } from "./rds-chart-line";
export { default as RdsPieChart } from "./rds-chart-Pie";
export { default as RdsDoughnutChart } from "./rds-chart-doughnut";
export { default as RdsBarChart } from "./rds-chart-bar";
export { default as RdsBooleanChart } from "./rds-chart-boolean";
export { default as RdsMixedChart } from "./rds-chart-mixed";
export { default as RdsPolarAreaChart } from "./rds-chart-polar-area";
export { default as RdsRadarChart } from "./rds-chart-radar";
export { default as RdsScatterChart } from "./rds-chart-scatter";
export { default as RdsStackedChart } from "./rds-chart-stacked";
export { default as RdsBubbleChart } from "./rds-chart-bubble";
export { default as RdsFeed } from "./rds-feed";
export { default as RdsCardDetail } from "./rds-card-detail";
// //Map
export { default as RdsMap } from "./rds-map";
export { default as RdsLoader } from "./rds-loader";
