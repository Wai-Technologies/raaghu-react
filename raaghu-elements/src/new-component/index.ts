export {default} from "./rds-comp-account";