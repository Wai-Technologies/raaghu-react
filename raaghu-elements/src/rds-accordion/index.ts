export { default } from "./rds-accordion";
export { default as RdsAccordionItem } from "./rds-accordion-item";