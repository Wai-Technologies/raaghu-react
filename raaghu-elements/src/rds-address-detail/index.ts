export {default} from "./rds-address-detail";