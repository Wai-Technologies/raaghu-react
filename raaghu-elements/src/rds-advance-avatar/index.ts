export { default } from "./rds-advance-avatar";