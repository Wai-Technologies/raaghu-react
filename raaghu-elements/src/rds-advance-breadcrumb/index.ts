export { default } from "./rds-advance-breadcrumb";