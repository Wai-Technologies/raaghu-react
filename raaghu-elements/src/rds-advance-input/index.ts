export { default } from "./rds-advance-input";