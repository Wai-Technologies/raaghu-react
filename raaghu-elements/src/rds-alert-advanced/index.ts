export { default } from "./rds-alert-advanced";