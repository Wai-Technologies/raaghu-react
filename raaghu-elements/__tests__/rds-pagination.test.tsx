import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { RdsPagination } from "../src";
import '@testing-library/jest-dom';


jest.mock('lottie-web')
jest.mock('react-lottie-player', () => ({
    __esModule: true,
    default: jest.fn(),
  }));

   
// Mock the useTranslation hook
jest.mock("react-i18next", () => ({
    useTranslation: () => ({ t: (key: string) => key }),
  }));

describe("RdsPagination", () => {
    it("renders correctly with default props", () => {
        const { container } = render(<RdsPagination totalRecords={100} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test("renders the correct number of pages based on total records and records per page", () => {
        const { getByTestId, getByText } = render(
            <RdsPagination totalRecords={50} recordsPerPage={10} />
        );
        expect(getByTestId("page-link")).toHaveTextContent("12345");
        fireEvent.click(getByText("2"));
        expect(getByTestId("page-link")).toHaveTextContent("12345");
    });

});