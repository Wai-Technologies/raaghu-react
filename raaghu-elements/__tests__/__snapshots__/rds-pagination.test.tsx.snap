// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RdsPagination renders correctly with default props 1`] = `
<div
  data-testid="page-link"
>
  <nav
    aria-label="Page navigation example"
  >
    <ul
      class="pagination align-items-center mb-0 pagination-sm pagination justify-content-start"
    >
      <li
        class="page-item cursor-pointer disabled d-none"
      >
        <a
          class="page-link b d-flex border-0"
        >
          <span
            class="ms-1 pagination-prev"
          >
            Prev
          </span>
        </a>
      </li>
      <li
        class="page-item m-1 cursor-pointer  "
      >
        <a
          class="page-link b d-flex border-0"
        >
          <span
            class="pagination-next me-1"
          >
            Next
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
